content/docs/guides/dockerswarm.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Docker Swarm
+sort_rank: 1
+---
+
+# Docker Swarm
+
+Prometheus can discover targets in a [Docker Swarm][swarm] cluster, as of
+v2.20.0. This guide demonstrates how to use that service discovery mechanism.
+
+## Docker Swarm service discovery architecture
+
+The [Docker Swarm service discovery][swarmsd] contains 3 different roles: nodes, services,
+and tasks.
+
+The first role, **nodes**, represents the hosts that are part of the Swarm. It
+can be used to automatically monitor the Docker daemons or the Node Exporters
+who run on the Swarm hosts.
+
+The second one, **services**, will discover the services deployed in the
```

Comment:
This  should probably be listed last. As worded it'd seem as though this is the right way to do things generally, rather than being a niche thing for blackbox use.

Reply:
I have reworded this